ntfy/cli.py

```diff
@@ -44,16 +46,58 @@ def run_cmd(args):
         *map(int, divmod(time() - start_time, 60))
     )
 
+emojis_hash = {}
+try:
+    emojis_json_path = join(dirname(getfile(run_cmd)),
```

Comment:
Force pushed update with the new emoji library to dan/emoji

Still having some issues with the tests (under py3, still had to pip
install --upgrade mock)
Also, haven't tried "python setup.py test" after doing build/install.

Vanilla works - py2 before doing any other setup.py (build/install).  No
pip install mock this time:

dan@phobos:~/src/myfork/ntfy$ mkvirtualenv --system-site-packages dantestpy2
New python executable in dantestpy2/bin/python
Installing setuptools, pip, wheel...done.
virtualenvwrapper.user_scripts creating
/home/dan/.virtualenvs/dantestpy2/bin/predeactivate
virtualenvwrapper.user_scripts creating
/home/dan/.virtualenvs/dantestpy2/bin/postdeactivate
virtualenvwrapper.user_scripts creating
/home/dan/.virtualenvs/dantestpy2/bin/preactivate
virtualenvwrapper.user_scripts creating
/home/dan/.virtualenvs/dantestpy2/bin/postactivate
virtualenvwrapper.user_scripts creating
/home/dan/.virtualenvs/dantestpy2/bin/get_env_details

(dantestpy2)dan@phobos:~/src/myfork/ntfy$ python setup.py test
/home/dan/.virtualenvs/dantestpy2/local/lib/python2.7/site-packages/setuptools/dist.py:285:
UserWarning: Normalizing '0.5.1-43' to '0.5.1.post43'
  normalized_version,
running test
Searching for mock
Best match: mock 1.3.0
Processing mock-1.3.0-py2.7.egg

Using /home/dan/src/myfork/ntfy/.eggs/mock-1.3.0-py2.7.egg
Searching for funcsigs
Best match: funcsigs 0.4
Processing funcsigs-0.4-py2.7.egg

Using /home/dan/src/myfork/ntfy/.eggs/funcsigs-0.4-py2.7.egg
running egg_info
writing requirements to ntfy.egg-info/requires.txt
writing ntfy.egg-info/PKG-INFO
writing top-level names to ntfy.egg-info/top_level.txt
writing dependency_links to ntfy.egg-info/dependency_links.txt
writing entry points to ntfy.egg-info/entry_points.txt
reading manifest file 'ntfy.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'ntfy.egg-info/SOURCES.txt'
running build_ext
test_backwards_compat (tests.ntfy_test.cli.TestLoadConfig) ... ok
test_default_config (tests.ntfy_test.cli.TestLoadConfig) ... ok
test_darwin (tests.ntfy_test.integration.TestIntegration) ... ok
test_default (tests.ntfy_test.integration.TestIntegration) ... ok
test_linux (tests.ntfy_test.integration.TestIntegration) ... ok
test_pushbullet (tests.ntfy_test.integration.TestIntegration) ... ok
test_pushover (tests.ntfy_test.integration.TestIntegration) ... ok
test_win32 (tests.ntfy_test.integration.TestIntegration) ... ok

----------------------------------------------------------------------
Ran 8 tests in 0.042s

OK

same but python3:

(dantestpy2)dan@phobos:~/src/myfork/ntfy$ deactivate
dan@phobos:~/src/myfork/ntfy$ mkvirtualenv --python python3
--system-site-packages dantestpy3
Running virtualenv with interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in dantestpy3/bin/python3
Also creating executable in dantestpy3/bin/python
Installing setuptools, pip, wheel...done.
virtualenvwrapper.user_scripts creating
/home/dan/.virtualenvs/dantestpy3/bin/predeactivate
virtualenvwrapper.user_scripts creating
/home/dan/.virtualenvs/dantestpy3/bin/postdeactivate
virtualenvwrapper.user_scripts creating
/home/dan/.virtualenvs/dantestpy3/bin/preactivate
virtualenvwrapper.user_scripts creating
/home/dan/.virtualenvs/dantestpy3/bin/postactivate
virtualenvwrapper.user_scripts creating
/home/dan/.virtualenvs/dantestpy3/bin/get_env_details

(dantestpy3)dan@phobos:~/src/myfork/ntfy$ python setup.py test
/home/dan/.virtualenvs/dantestpy3/lib/python3.4/site-packages/setuptools/dist.py:285:
UserWarning: Normalizing '0.5.1-43' to '0.5.1.post43'
  normalized_version,
running test
Searching for emoji
Best match: emoji 0.3.9
Processing emoji-0.3.9-py3.4.egg

Using /home/dan/src/myfork/ntfy/.eggs/emoji-0.3.9-py3.4.egg
Searching for sleekxmpp
Best match: sleekxmpp 1.3.1
Processing sleekxmpp-1.3.1-py3.4.egg

Using /home/dan/src/myfork/ntfy/.eggs/sleekxmpp-1.3.1-py3.4.egg
Searching for mock
Best match: mock 1.3.0
Processing mock-1.3.0-py3.4.egg

Using /home/dan/src/myfork/ntfy/.eggs/mock-1.3.0-py3.4.egg
Traceback (most recent call last):
  File "setup.py", line 70, in <module>
    'ntfy = ntfy.cli:main',
  File "/usr/lib/python3.4/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.4/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File
"/home/dan/.virtualenvs/dantestpy3/lib/python3.4/site-packages/setuptools/command/test.py",
line 134, in run
    self.distribution.fetch_build_eggs(self.distribution.tests_require)
  File
"/home/dan/.virtualenvs/dantestpy3/lib/python3.4/site-packages/setuptools/dist.py",
line 313, in fetch_build_eggs
    replace_conflicting=True,
  File
"/home/dan/.virtualenvs/dantestpy3/lib/python3.4/site-packages/pkg_resources/__init__.py",
line 836, in resolve
    dist = best[req.key] = env.best_match(req, ws, installer)
  File
"/home/dan/.virtualenvs/dantestpy3/lib/python3.4/site-packages/pkg_resources/__init__.py",
line 1074, in best_match
    dist = working_set.find(req)
  File
"/home/dan/.virtualenvs/dantestpy3/lib/python3.4/site-packages/pkg_resources/__init__.py",
line 711, in find
    raise VersionConflict(dist, req)
pkg_resources.VersionConflict: (six 1.5.2 (/usr/lib/python3/dist-packages),
Requirement.parse('six>=1.7'))

so:

(dantestpy3)dan@phobos:~/src/myfork/ntfy$ pip install --upgrade mock
Collecting mock
  Using cached mock-1.3.0-py2.py3-none-any.whl
Collecting six>=1.7 (from mock)
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting pbr>=0.11 (from mock)
  Using cached pbr-1.8.1-py2.py3-none-any.whl
Installing collected packages: six, pbr, mock
  Found existing installation: six 1.5.2
    DEPRECATION: Uninstalling a distutils installed project (six) has been
deprecated and will be removed in a future version. This is due to the fact
that uninstalling a distutils project will only partially uninstall the
project.
    Not uninstalling six at /usr/lib/python3/dist-packages, outside
environment /home/dan/.virtualenvs/dantestpy3
Successfully installed mock-1.3.0 pbr-1.8.1 six-1.10.0
You are using pip version 7.1.2, however version 8.0.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
(dantestpy3)dan@phobos:~/src/myfork/ntfy$ python setup.py test
/home/dan/.virtualenvs/dantestpy3/lib/python3.4/site-packages/setuptools/dist.py:285:
UserWarning: Normalizing '0.5.1-43' to '0.5.1.post43'
  normalized_version,
running test
Searching for emoji
Best match: emoji 0.3.9
Processing emoji-0.3.9-py3.4.egg

Using /home/dan/src/myfork/ntfy/.eggs/emoji-0.3.9-py3.4.egg
Searching for sleekxmpp
Best match: sleekxmpp 1.3.1
Processing sleekxmpp-1.3.1-py3.4.egg

Using /home/dan/src/myfork/ntfy/.eggs/sleekxmpp-1.3.1-py3.4.egg
running egg_info
writing requirements to ntfy.egg-info/requires.txt
writing entry points to ntfy.egg-info/entry_points.txt
writing ntfy.egg-info/PKG-INFO
writing dependency_links to ntfy.egg-info/dependency_links.txt
writing top-level names to ntfy.egg-info/top_level.txt
reading manifest file 'ntfy.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'ntfy.egg-info/SOURCES.txt'
running build_ext
test_backwards_compat (tests.ntfy_test.cli.TestLoadConfig) ... ok
test_default_config (tests.ntfy_test.cli.TestLoadConfig) ...
/home/dan/src/myfork/ntfy/ntfy/cli.py:24: ResourceWarning: unclosed file
<_io.TextIOWrapper name='/dev/null' mode='r' encoding='UTF-8'>
  config = {'backends': ['default']}
ok
test_darwin (tests.ntfy_test.integration.TestIntegration) ... ok
test_default (tests.ntfy_test.integration.TestIntegration) ... ok
test_linux (tests.ntfy_test.integration.TestIntegration) ... ok
test_pushbullet (tests.ntfy_test.integration.TestIntegration) ... ok
test_pushover (tests.ntfy_test.integration.TestIntegration) ... ok
test_win32 (tests.ntfy_test.integration.TestIntegration) ... ok

----------------------------------------------------------------------
Ran 8 tests in 0.125s

OK


On Tue, Jan 19, 2016 at 9:46 PM, Daniel Schep <notifications@github.com>
wrote:

> In ntfy/cli.py
> <https://github.com/dschep/ntfy/pull/10#discussion_r50208755>:
>
> > @@ -44,16 +46,58 @@ def run_cmd(args):
> >          *map(int, divmod(time() - start_time, 60))
> >      )
> >
> > +emojis_hash = {}
> > +try:
> > +    emojis_json_path = join(dirname(getfile(run_cmd)),
>
> Oh wait, you're switching to https://pypi.python.org/pypi/emoji so it
> doesn't matter.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dschep/ntfy/pull/10/files#r50208755>.
>